pyannote/core/annotation.py

```diff
@@ -286,9 +301,17 @@ def itertracks(
                 tracks.items(), key=lambda tl: (str(tl[0]), str(tl[1]))
             ):
                 if yield_label:
-                    yield segment, track, lbl
+                    yield SegmentTrackLabel(segment, track, lbl)
                 else:
-                    yield segment, track
+                    yield SegmentTrack(segment, track)
+
+    def itertracks_with_labels(self) -> Iterator[SegmentTrackLabel]:
+        """Typed version of :func:`itertracks`(yield_label=True)"""
+        return self.itertracks_with_labels()  # type: ignore
```

Comment:
I think this is what you meant:

```suggestion
        return self.itertracks(yield_label=True)  # type: ignore
```

However—while I'd be fine with having `itertracks_with_labels` and `itertracks_without_labels`—a better approach to getting typed `itertracks` might be using `overload` like I did [here](https://github.com/evfinkn/pyannote-core/commit/d981be262d050fa13d19c4910fbae14c49a4d118):

```python
    @overload
    def itertracks(self, yield_label: Literal[False] = ...) -> Iterator[Track]: ...
    @overload
    def itertracks(self, yield_label: Literal[True]) -> Iterator[LabeledTrack]: ...
    @overload
    def itertracks(self, yield_label: bool) -> TrackIterator: ...
```

(Though this would obviously need to be slightly altered to use your `NamedTuples`). This has the added benefit of working with existing code. The same thing can probably be done with `Timeline.crop`.

Reply:
Thanks for the comments @evfinkn 
I have tried overloaded typing with bool constants in the past, and then stumbled upon the note that overloading is only allowed for types, not values. However, I did not consider your approach of using `Literal[False]`. I will have to see if this works with pylance. if it does the overload approach is far more elegant 😀